### PR TITLE
Avoid extra buffer to prepend frame size

### PR DESCRIPTION
**Pull request by merlimat**
### Motivation

As described in #3559, we should avoid the extra buffer allocation just for prepending the 4 bytes frame size

### CODE REVIEW

**Review by hangc0276 — APPROVED** [review]

**Review by eolivelli — APPROVED** [review]
LGTM

**Comment by eolivelli** [comment]
please take a look to tests

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieProtocol.java [hunk 1]
```diff
@@ -481,12 +482,14 @@ public ReferenceCounted retain(int increment) {
 
         @Override
         public ReferenceCounted touch() {
-            return data.touch();
```
Comment (merlimat) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieProtocol.java:484]: When using the leak checker, this was causing the `ReadResponse` to get converted into an instrumented buffer, and the serialization was completely skipped.

**Review by zymap — APPROVED** [review]

**Comment by hangc0276** [comment]
@merlimat Would you please take a look at those failed tests?

**Pull request merged**